### PR TITLE
clarify Raft usage

### LINKED ISSUE #195: Update Raft documentation to indicate when and when it isn't suitable for use
Raft consensus is not suitable for production networks.  Update documentation to make this clear and that it should at most be used for development environments.

### PR DESCRIPTION
**Pull request by alexandratran**
# Pull request checklist

Use the following template to make sure your PR fits the GoQuorum documentation standard.

## Before creating the PR

Make sure that:

- [x] You've read the [contribution guidelines](https://consensys.net/docs/doctools/).
- [x] You've [previewed your changes locally](https://consensys.net/docs/doctools/en/latest/preview/new-system/#preview-locally).

## Describe the change

Add warnings about using Raft only in development environments.

## Issue fixed

fixes #195 

## Impacted parts

For content changes:

- [x] Documentation content
- [ ] Documentation page organization

For tool changes:

- [ ] CircleCI workflow
- [ ] Build and QA tools (e.g., lint or vale)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] Read the Docs configuration
- [ ] GitHub integration

## After creating your PR and tests have finished

Make sure that:

- [x] You've [fixed any issues](https://consensys.net/docs/doctools/en/latest/contribute/fix-cicd-errors/) raised by the tests.
- [x] You've [previewed your changes on GitHub Pages](https://consensys.net/docs/doctools/en/latest/preview/new-system/#preview-on-github-pages)

### CODE REVIEW

File: docs/concepts/consensus/overview.md [hunk 1]
```diff
@@ -9,7 +9,10 @@ GoQuorum implements the following Proof of Authority consensus protocols:
 
 !!! warning
 
-    QBFT is currently an early access feature. It is not recommended for production networks with business critical impact.
+    - QBFT is currently an early access feature. It is not recommended for production networks with business critical impact.
```
Comment (NicolasMassart) [inline docs/concepts/consensus/overview.md:12]: I understood that QBFT was ready, maybe we should remove this line. But please confirm with GoQuorum team.

Comment (joshuafernandes) [reply docs/concepts/consensus/overview.md:12]: @antonydenyer ^

Comment (joshuafernandes) [reply docs/concepts/consensus/overview.md:12]: I think this one is ready @NicolasMassart, @baptiste-b-pegasys raised #188 for this as well

Comment (MadelineMurray) [reply docs/concepts/consensus/overview.md:12]: @antonydenyer - can we remove this line in GoQuorum now.  Lucas thought so but it's a GoQuorum question not a QBFT question.

File: docs/configure-and-manage/configure/consensus-protocols/raft.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ GoQuorum implements the Raft Proof-of-Authority (PoA) consensus protocol.
 To enable Raft consensus, specify the [`--raft`](../../../reference/cli-syntax.md#raft) command line option when starting GoQuorum.
 You can [create a private network using Raft](../../../tutorials/private-network/create-a-raft-network.md).
 
+!!! warning
+
+    Raft is not suitable for production environments. Use only in development environments. Also, you can't migrate to
+    another consensus protocol from Raft.
```
Comment (MadelineMurray) [inline docs/configure-and-manage/configure/consensus-protocols/raft.md:14]: We need to add a bit more nuance to the migration point: 

Migrating from a network from Raft to a consensus mechanism suitable for production such as QBFT requires one of: 
* Stopping the Raft network and starting the new network with the state at the time of migration.  That is, historical transactions and state history are lost. 
* Replying the historical transactions on the new network.  The historical transactions will be a different block heights but other than that, the transactions and state history will be the same on the new network as on the Raft network. 

If you want Enterprise support to migrate a Raft network, contact us (link to relevant support contact).

**Review by antonydenyer — APPROVED** [review]

File: docs/configure-and-manage/configure/consensus-protocols/raft.md [hunk 3]
```diff
@@ -45,3 +51,16 @@ By default, GoQuorum listens on port 50400 for the Raft transport. Use the
 
 By default, the number of peers is 25. Use the `--maxpeers N` command line option to configure the
 maximum number of peers where N is expected size of the cluster.
+
+## Migrate from Raft to another consensus protocol
+
+Migrating a network using Raft to a consensus mechanism suitable for production such as [QBFT](qbft.md) requires one of:
```
Comment (rolandtyler) [inline docs/configure-and-manage/configure/consensus-protocols/raft.md:57]:
```suggestion
Migrating a network using Raft to a consensus mechanism suitable for production such as [QBFT](qbft.md) requires one of the following:
```

**Review by rolandtyler — APPROVED** [review]
one comment, otherwise lgtm.

File: docs/configure-and-manage/configure/consensus-protocols/raft.md [hunk 4]
```diff
@@ -45,3 +51,16 @@ By default, GoQuorum listens on port 50400 for the Raft transport. Use the
...
 maximum number of peers where N is expected size of the cluster.
+
+## Migrate from Raft to another consensus protocol
+
+Migrating a network using Raft to a consensus mechanism suitable for production such as [QBFT](qbft.md) requires one of the following:
+
+* Stopping the Raft network and starting the new network with the state at the time of migration.
+  That is, historical transactions and state history are lost.
+
+* Replaying the historical transactions on the new network.
```
Comment (bgravenorst) [inline docs/configure-and-manage/configure/consensus-protocols/raft.md:62]: Do we describe anywhere how to do this? Might be worth linking too. Or perhaps raise create a ticket to add the missing content

File: docs/configure-and-manage/configure/consensus-protocols/raft.md [hunk 5]
```diff
@@ -45,3 +51,16 @@ By default, GoQuorum listens on port 50400 for the Raft transport. Use the
...
+  The historical transactions will be at different block heights, but otherwise the transactions and state history will
+  be the same on the new network as on the Raft network.
```
Comment (bgravenorst) [inline docs/configure-and-manage/configure/consensus-protocols/raft.md:64]:
```suggestion
  The historical transactions will be at different block heights, but the transactions and state history will
  be the same on the new network as on the Raft network.
```

File: docs/configure-and-manage/configure/consensus-protocols/raft.md [hunk 6]
```diff
@@ -45,3 +51,16 @@ By default, GoQuorum listens on port 50400 for the Raft transport. Use the
 
 By default, the number of peers is 25. Use the `--maxpeers N` command line option to configure the
 maximum number of peers where N is expected size of the cluster.
+
+## Migrate from Raft to another consensus protocol
+
+Migrating a network using Raft to a consensus mechanism suitable for production such as [QBFT](qbft.md) requires one of the following:
```
Comment (bgravenorst) [inline docs/configure-and-manage/configure/consensus-protocols/raft.md:57]: Should we raise a ticket to add a tutorial for this?

File: docs/concepts/consensus/overview.md [hunk 7]
```diff
@@ -9,7 +9,9 @@ GoQuorum implements the following Proof of Authority consensus protocols:
 
 !!! warning
 
-    QBFT is currently an early access feature. It is not recommended for production networks with business critical impact.
+    Raft is not suitable for production environments.
+    Use only in development environments.
+    You can [migrate a network using Raft to another consensus protocol](../../configure-and-manage/configure/consensus-protocols/raft.md#migrate-from-raft-to-another-consensus-protocol).
```
Comment (bgravenorst) [inline docs/concepts/consensus/overview.md:14]:
```suggestion
    You can [migrate a Raft network to another consensus protocol](../../configure-and-manage/configure/consensus-protocols/raft.md#migrate-from-raft-to-another-consensus-protocol).
```

File: docs/configure-and-manage/configure/consensus-protocols/raft.md [hunk 8]
```diff
@@ -8,6 +8,12 @@ GoQuorum implements the Raft Proof-of-Authority (PoA) consensus protocol.
 To enable Raft consensus, specify the [`--raft`](../../../reference/cli-syntax.md#raft) command line option when starting GoQuorum.
 You can [create a private network using Raft](../../../tutorials/private-network/create-a-raft-network.md).
 
+!!! warning
+
+    Raft is not suitable for production environments.
+    Use only in development environments.
+    You can [migrate a network using Raft to another consensus protocol](#migrate-from-raft-to-another-consensus-protocol).
```
Comment (bgravenorst) [inline docs/configure-and-manage/configure/consensus-protocols/raft.md:15]:
```suggestion
    You can [migrate a Raft network to another consensus protocol](#migrate-from-raft-to-another-consensus-protocol).
```

File: docs/tutorials/private-network/create-a-raft-network.md [hunk 9]
```diff
@@ -7,7 +7,14 @@ description: Creating a network using Raft consensus
 A private network provides a configurable network for testing. This private network uses the
 [Raft consensus protocol](../../configure-and-manage/configure/consensus-protocols/raft.md).
 
-!!!important
+!!! warning
+
+    Raft is not suitable for production environments.
+    Use only in development environments.
+    You can [migrate a network using Raft to another consensus protocol](../../configure-and-manage/configure/consensus-protocols/raft.md).
```
Comment (bgravenorst) [inline docs/tutorials/private-network/create-a-raft-network.md:14]:
```suggestion
    You can [migrate a Raft network to another consensus protocol](../../configure-and-manage/configure/consensus-protocols/raft.md).
```

**Review by bgravenorst — APPROVED** [review]
Some feedback/questions

File: docs/configure-and-manage/configure/consensus-protocols/raft.md (hunk 4 above)
Comment (alexandratran) [reply docs/configure-and-manage/configure/consensus-protocols/raft.md:62]: Antony says it depends on the user's specific network setup and situtation, so I think just advising to contact support is fine.

**Pull request merged**